lib/services/AutoRenewService.ts

```diff
@@ -0,0 +1,85 @@
+/*!
+ * Copyright (c) 2015-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * 
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+
+import { TokenManager, EVENT_EXPIRED } from '../TokenManager';
+import { AuthSdkError } from '../errors';
+import { ServiceInterface, ServiceManagerOptions } from '../types';
+import { isBrowser } from '../features';
+
+export class AutoRenewService implements ServiceInterface {
+  private tokenManager: TokenManager;
+  private options: ServiceManagerOptions;
+  private renewTimeQueue: Array<number>;
+  private started = false;
+
+  constructor(tokenManager: TokenManager, options: ServiceManagerOptions = {}) {
```

Comment:
pass `oktaAuth` in constructor, so both serviceManager and tokenManager can be available in each service class.